java/src/jmri/jmrix/sprog/SprogProgrammer.java

```diff
@@ -134,12 +174,17 @@ protected void useProgrammer(jmri.ProgListener p) throws jmri.ProgrammerExceptio
      * @param mode Mode to be used
      * @param val value to be written
      * @param cvnum CV address to write to 
+     * @param startVal Hint of what the CV may contain
      * @return formatted message to do programming operation
      */
-    protected SprogMessage progTaskStart(ProgrammingMode mode, int val, int cvnum) {
+    protected SprogMessage progTaskStart(ProgrammingMode mode, int val, int cvnum, int startVal) {
         // val = -1 for read command; mode is direct, etc
         if (val < 0) {
-            return SprogMessage.getReadCV(cvnum, mode);
+            if (startVal == NO_DEFAULT_CV) {
+                return SprogMessage.getReadCV(cvnum, mode);
+            } else {
+                return SprogMessage.getReadCV(cvnum, mode, startVal);
+            }
```

Comment:
You don't need this `if` statement since `SprogMessage.readCV(_cv, this, NO_DEFAULT_CV)` is the same as `SprogMessage.readCV(_cv, this)`.